src/Parser/FireMage/Modules/Features/AlwaysBeCasting.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import CoreAlwaysBeCasting from 'Parser/Core/Modules/AlwaysBeCasting';
+
+import SPELLS from 'common/SPELLS';
+import Icon from 'common/Icon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import SpellLink from 'common/SpellLink';
+
+class AlwaysBeCasting extends CoreAlwaysBeCasting {
+  static ABILITIES_ON_GCD = [
+    SPELLS.FIREBALL.id,
+    SPELLS.PYROBLAST.id,
+    SPELLS.BLAZING_BARRIER.id,
+    SPELLS.FROST_NOVA.id,
+    SPELLS.BLINK.id,
+    SPELLS.ICE_BLOCK.id,
+    SPELLS.FLAMESTRIKE.id,
+    SPELLS.SCORCH.id,
+    SPELLS.PHOENIXS_FLAMES.id,
+    SPELLS.DRAGONS_BREATH.id,
+    // talents
+    SPELLS.BLAST_WAVE_TALENT.id,
+    SPELLS.MIRROR_IMAGE_TALENT.id,
+    SPELLS.RUNE_OF_POWER_TALENT.id,
+    SPELLS.RING_OF_FROST_TALENT.id,
+    SPELLS.LIVING_BOMB_TALENT.id,
+    SPELLS.METEOR_TALENT.id,
+    SPELLS.CINDERSTORM_TALENT.id,
+  ];
+
+  suggestions(when) {
+    const deadTimePercentage = this.totalTimeWasted / this.owner.fightDuration;
+
+    when(deadTimePercentage).isGreaterThan(0.2)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your downtime can be improved. Try to Always Be Casting (ABC), try to reduce the delay between casting spells. Even if you have to move, try casting instants, even unbuffed <SpellLink id={SPELLS.ICE_LANCE_CAST.id} /> spam is better than nothing.</span>)
```

Comment:
Fix the string to be Fire Mage relevant. Probably all you need to do is swap in Scorch for Ice Lance.